### PR TITLE
Use numeric_cast_v<std::size_t> instead of deprecated integer2size_t [blocks: #2310]

### PR DESCRIPTION
**Pull request by tautschnig**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by tautschnig** [comment]
~I will revisit this once #3047 is merged (they have conflicting/duplicating changes).~

File: src/goto-programs/goto_convert.cpp [hunk 1]
```diff
@@ -1903,7 +1903,7 @@ bool goto_convertt::get_string_constant(
       forall_operands(it, index_op)
         if(it->is_constant())
         {
-          unsigned long i=integer2ulong(
+          const std::size_t i = numeric_cast_v<std::size_t>(
             binary2integer(id2string(to_constant_expr(*it).get_value()), true));
```
Comment (kroening) [inline src/goto-programs/goto_convert.cpp]: BTW, just ran into that one again (in #3100).

I'd advocate removing this method, given that I don't think it works.

Comment (tautschnig) [reply src/goto-programs/goto_convert.cpp]: Which method are you referring to? I wasn't able to figure out which problem #3100 is solving, so I'd need more info. Thanks!

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: fd6116968dc021445bdceb913eefb25d18d3e7ed).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87937825
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 9cd404b4adec54b5fb090a77faef6660d15ea773).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87938549

**Review by martin-cs — APPROVED** [review]
This change appears to be as described in all the areas I am "owner" of.

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: d73108e16c83ee1cb3d06aade03790777db53b78).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88736803

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 23ff9c9ec4adf7516248d810b9db6a0f661be80e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89875998

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 7a4c5e1fe3266b3d351cfb0cec80ea2d86a84d78).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90159583
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by martin-cs** [comment]
How does this relate to https://github.com/diffblue/cbmc/pull/3252 ?  Should this be updated / closed in light of that?

**Comment by tautschnig** [comment]
> How does this relate to #3252 ? Should this be updated / closed in light of that?

No, they refer to two different types being converted to: this one is for `std::size_t`, #3252 was `mp_integer`.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5d837c0997818befb8da7a3a6bb6fd31a2453c17).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90496251

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: eca50ceaeebef0d8ed58a39ebd056c32d584888b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90620816
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:warning:
Diffblue compatibility check is currently unavailable.
Please create manual bump.
(cbmc commit: fe3f73c7a8ec52fc5a9ecbd08660694bb3b92e59).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90844662

**Comment by romainbrenguier** [comment]
Check actually passed :heavy_check_mark:

**Pull request merged**